librad/src/git/mod.rs

```diff
@@ -266,7 +266,7 @@ where
             .map(|binding| binding.userid())?;
 
         // FIXME: use `Option::flatten` once out of nightly
-        let addr = if let Ok(Some(addr)) = uid.address() {
+        let addr = if let Ok(Some(addr)) = uid.email() {
```

Comment:
You could do `uid.email().and_then(|addr| addr.ok_or(Error::MissingPgpAddr))`, possibly?